lib/clouds/libcloud_common.py

```diff
@@ -571,166 +610,329 @@ def get_vm_instance(self, obj_attr_list) :
-        obj_attr_list["cloud_vm_uuid"] = "NA"
-        _instance = False
-        volume = False
-        extra = deepcopy(self.extra)
-        obj_attr_list["cloud_vm_name"] = "cb-" + obj_attr_list["username"]
-        obj_attr_list["cloud_vm_name"] += '-' + "vm" + obj_attr_list["name"].split("_")[1]
-        obj_attr_list["cloud_vm_name"] += '-' + obj_attr_list["role"]
+    def is_vm_running(self, obj_attr_list):
+        '''
+        TBD
+        '''
+        try :
+            node = self.get_instances(obj_attr_list)
+            return node and node.state == NodeState.RUNNING
+
+        except Exception, e :
+            _status = 23
+            _fmsg = str(e)
+            raise CldOpsException(_fmsg, _status)
+
+    @trace
+    def is_vm_ready(self, obj_attr_list) :
+        '''
+        TBD
+        '''
+        if self.is_vm_running(obj_attr_list) :
+
+            if self.get_ip_address(obj_attr_list) :
+                obj_attr_list["last_known_state"] = "running with ip assigned"
+                return True
+            else :
+                obj_attr_list["last_known_state"] = "running with ip unassigned"
+                return False
+        else :
+            obj_attr_list["last_known_state"] = "not running"
+            return False
 
+    @trace
+    def vm_placement(self, obj_attr_list) :
+        '''
+        TBD
+        '''
         try :
+            _status = 100
+            _fmsg = "An error has occurred, but no error message was captured"
 
-            obj_attr_list["host_name"] = obj_attr_list["vmc_name"]
+            _status = 0
 
-            if obj_attr_list["ai"] != "none" :
-                obj_attr_list["cloud_vm_name"] += '-' + obj_attr_list["ai_name"]
+        except Exception, e :
+            _status = 23
+            _fmsg = str(e)
+            
+        finally :
+            if _status :
+                _msg = "VM placement failed: " + _fmsg
+                cberr(_msg, True)
+                raise CldOpsException(_msg, _status)
+            else :
+                return True
+
+    @trace
+    def vvcreate(self, obj_attr_list, connection) :
+        '''
+        TBD
+        '''
+        try :
+            _status = 100
+            _fmsg = "An error has occurred, but no error message was captured"
+
+            obj_attr_list["cloud_vv_instance"] = None
+
+            if "cloud_vv_type" not in obj_attr_list :
+                obj_attr_list["cloud_vv_type"] = "DOC"
+
+            if "cloud_vv" in obj_attr_list :
+
+                obj_attr_list["cloud_vv_name"] = obj_attr_list["cloud_vv_name"].lower()
+                
+                obj_attr_list["last_known_state"] = "about to send volume create request"
+
+                self.common_messages("VV", obj_attr_list, "creating", _status, _fmsg)
+                
+                _mark1 = int(time())
+
+                _volume = connection.create_volume(int(obj_attr_list["cloud_vv"]),
```

Comment:
Alright, so the volume conditionals need to be brought back. The purpose of those conditionals was for clouds like VCloud whose clouds don't support volumes at all within the libcloud driver for each particular cloud. This needs to be hard-coded and has no bearing on CLOUD_VV=X.